### PR TITLE
Merge the RadMon data extract source code into rrfs_util.

### PR DESCRIPTION
**Pull request by xyzemc**
https://github.com/NOAA-GSL/rrfs_utl/issues/43

The RadMon (radiance monitoring) package is used to extract data from
radiance diagnostic files to generate the binary files for plotting the statistics from radiance data assimilation, and generate tons of images for each satellite instrument and channel.  It has been used in GDAS and NAM DA operational.
This pull request is for adding the RadMon source code into rrfs_util.

The data extract step from RadMon include four packs of source code:
verf_radang.fd/  
verf_radbcoef.fd/  
verf_radbcor.fd/ 
verf_radtime.fd/

All these codes have put in the directory 'radmon' under rrfs_util. The corresponding makefile has been modified and tested.  Four executables will be placed into the ./bin directory as all other code did.

### CODE REVIEW

**Review by hu5970 — APPROVED** [review]

**Review by christopherwharrop-noaa — CHANGES_REQUESTED** [review]
There is a lot of duplication in this pull request.  The specification of compiler flags for different compilers should ideally be done in one place for the entire repository, not duplicated verbatim in multiple places.  Additionally, the introduction of multiple `kinds.f90` creates unnecessary duplication, particularly since rrfs_utls already has a kinds module.  Please remove duplication and make use of existing code and build specifications where possible.

**Comment by hu5970** [comment]
@christopherwharrop-noaa This repository is to host many small applications for RRFS. Some of those applications are isolated from others. I agree it is good to reduce the duplication. We also need to check this case by case. "kind.f90" used here may not have the same definition as the kinds module used by other. Do we have a way to keep some code used locally (like kinds.f90 only used by this application)? The same possible is also applying for compiling options.

**Comment by xyzemc** [comment]
@christopherwharrop-noaa @hu5970 I will compare the kind.f90 in this PR and the existing kind.f90 in rrfs_util to see if they are same and can avoid the duplication.

**Comment by hu5970** [comment]
@xyzemc Please remove save directory and sync with current develop branch.

File: radmon/shared/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  read_diag.f90
+  sparsearr.f90
+)
+
+set(module_dir "${CMAKE_CURRENT_BINARY_DIR}/include")
+add_library(radmonlib STATIC ${SRCS})
+add_library(${PROJECT_NAME}::radmonlib ALIAS radmonlib)
+
+target_compile_definitions(radmonlib PRIVATE "_REAL8_")
+#target_link_libraries(radmonlib PUBLIC ${PROJECT_NAME}::ncio)
```
Comment (christopherwharrop-noaa) [inline radmon/shared/CMakeLists.txt]: Please remove commented code

File: radmon/verf_radbcoef.fd/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,17 @@
+
+list(APPEND SRCS
+  bcoef.f90
+  create_ctl_bcoef.f90
+)
+
+add_executable(radmon_bcoef.x ${SRCS})
+
+#include_directories( /scratch1/BMC/wrfruc/mhu/rrfs/v0.7.1/ufs-srweather-app/include )
```
Comment (christopherwharrop-noaa) [inline radmon/verf_radbcoef.fd/CMakeLists.txt]: Please remove commented code

File: radmon/verf_radtime.fd/bad_chan.f90 [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+      character(8), intent( in )	:: date
+      character(8), intent( in )	:: cycle
+      integer, intent( out )		:: fios     
+    
+      !--- variables
+      logical 				:: lexist = .FALSE. 
+      character(60)                     :: fname
+
+
+!      write(*,*) '--> open_bad_chan_file, date, cycle = ', date, cycle
```
Comment (christopherwharrop-noaa) [inline radmon/verf_radtime.fd/bad_chan.f90]: This looks like commented code leftover from debugging.  Perhaps it should be removed?

File: radmon/verf_radtime.fd/bad_penalty.f90 [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+      !--- interface
+      character(8), intent( in )	:: date
+      character(8), intent( in )	:: cycle
+      integer, intent( out )		:: fios     
+     
+      !--- variables
+      logical                           :: lexist = .FALSE.
+      character(60)                     :: fname
+      
+!      write(*,*) '--> open_bad_penalty_file, date, cycle = ', date, cycle
```
Comment (christopherwharrop-noaa) [inline radmon/verf_radtime.fd/bad_penalty.f90]: This looks like commented code leftover from debugging.  Perhaps it should be removed?

**Review by christopherwharrop-noaa — CHANGES_REQUESTED** [review]
Thank you for taking the time to address my comments.  This looks great!  There are a couple commented lines in `CMakeLists.txt` files that should be removed, and you might also consider removing what look like (I'm not sure) commented out debugging `write` statements.  Once the Cmake comments are removed, I'll approve.

**Review by christopherwharrop-noaa — APPROVED** [review]
Thank you for making those last few updates.

**Pull request merged**